app/controllers/verify/finance_controller.rb

```diff
@@ -17,8 +18,25 @@ def create
 
     private
 
+    def step
+      @_step ||= Idv::FinancialsStep.new(
+        idv_form: idv_finance_form,
+        idv_session: idv_session,
+        analytics: analytics,
+        params: step_params
+      )
+    end
+
+    def step_params
+      params.require(:idv_finance_form).permit(:finance_type, *Idv::FinanceForm::FINANCE_TYPES)
+    end
+
+    def confirm_step_needed
+      redirect_to verify_phone_path if idv_session.financials_confirmation.try(:success?)
+    end
```

Comment:
The purpose is both: one validation per session, and prevent the user from attempting to do more.

We only store the user's data in the DB once it is permanent. We don't ever store financial data in the DB because that is only used for IdV. We *want* the user to reset the state by signing out. We don't want to keep any non-permanent PII around past the immediate session.